src/utils/Store/Notifier.ts

```diff
@@ -0,0 +1,37 @@
+type Listener = () => void;
+
+export class Notifier {
+  private _listens: Listener[] = [];
+
+  get hasListen(): boolean {
+    return Boolean(this._listens.length);
+  }
+
+  addListen(listener: Listener): () => void {
+    const { _listens } = this;
+    _listens.push(listener);
+    return () => {
+      _listens.splice(_listens.indexOf(listener), 1);
+    };
+  }
+
+  protected notify(): void {
+    this._listens.slice().forEach((listener) => listener());
```

Comment:
We wonder why you need to `slice` here?

Reply:
it's will throw error when remove a listen in listener
ex: [listen1, listen2]
listen1 => remove listen2
run listen1 => array = [listen1] => missing listen2 but forEach is running.

but i trying compare performance of check undefined and slice to choose new solution.
Thank for ask